### ISSUE TITLE
Invalid png file makes stb_image hang

### ISSUE THREAD

**Issue by ikirill** [nothings/stb#757]
This isn't a valid PNG file, as seen from ImageMagick's output. The image was generated by https://llvm.org/docs/LibFuzzer.html
```
➜ fuzz_stb % convert timeout-b2397b7028472f3d9a4b772e9ff25f6e7d815965.png test.jpg
convert: IDAT: CRC error `timeout-b2397b7028472f3d9a4b772e9ff25f6e7d815965.png' @ error/png.c/MagickPNGErrorHandler/1714.
convert: no images defined `test.jpg' @ error/convert.c/ConvertImageCommand/3300.
```

When I load it with `stbi_load`, it just hangs instead of failing.

![timeout-b2397b7028472f3d9a4b772e9ff25f6e7d815965](https://user-images.githubusercontent.com/1158512/56744400-57e2f300-6770-11e9-9a13-f6f7db353345.png)

**Comment by zao** [comment]
Please note that the declared intent of stb_image is to read legitimate files authored under your control.

While there might be some interest in making functions more robust, handling a malicious adversary like a fuzzer is probably way out of scope.

**Comment by nothings** [comment]
It's not actually out of scope.

At this point stb_image use is widespread enough, and outside the limited-security-exposure of its original game-developer audience, that it is the one library we have been pretty serious about fixing (although it's current status is still problematic, because there was a time when I accepted any and all new image formats).

**Comment by zao** [comment]
@nothings Ah, sorry.
I went by prior experience and the note in the source file:
https://github.com/nothings/stb/blob/master/stb_image.h#L19-L21

**Comment by ikirill** [comment]
I read the line in https://github.com/nothings/stb/blob/master/.github/CONTRIBUTING.md saying "I generally do not want new file formats for stb_image because we are trying to improve its security"

**Comment by nothings** [comment]
> I went by prior experience and the note in the source file:
> https://github.com/nothings/stb/blob/master/stb_image.h#L19-L21

The note used to say it was **only** for when you controlled the images you were feeding into it. But over the 10+ years it's existed it became clear that I couldn't really prevent people from using it for everything, and taking a more security-minded approach was the only solution. So that note is a nod to the original intent without saying "only".

**Comment by rygorous** [comment]
This bug was fixed by commit bfaccab17a648b315543d366c63aee575a0756b7 which is part of the currently released version 2.26.
